framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -415,104 +334,113 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
+			const inclusionProof = ccmsListOfList[0][ccmsListOfList[0].length - 1].inclusionProof;
+
+			return [{
+				crossChainMessages: ccmHashesList,
+				messageWitnessHashes: [],
+				outboxRootWitness: inclusionProof,
+			}];
+		}
 
-		const validatorsHashPreimages =
-			await this._sidechainChainConnectorDB.getValidatorsHashPreImage();
-		const validatorData = validatorsHashPreimages.find(
-			data => data.validatorsHash === blockHeader.validatorsHash,
-		);
+		// Calculate list of inboxUpdates to be sent by multiple CCUs
+		const listOfInboxUpdates = [];
+		for (const subList of ccmsListOfList) {
+			const ccmHashesList = subList.map(ccm => codec.encode(ccmSchema, ccm));
+			// Take the inclusion proof of the last ccm height
+			const inclusionProof = subList[subList.length - 1].inclusionProof;
 
-		if (!validatorData) {
-			throw new Error('No Validators data found for the given validatorsHash.');
-		}
+			// Calculate message witnesses
 
-		for (let i = 0; i < validatorData.validators.length; i += 1) {
-			if (aggregateCommit.aggregationBits[i] === 1) {
-				const blsKey = validatorData.validators[i].blsKey.toString('hex');
-				if (!blsKeyToBFTWeight[blsKey]) {
-					return false;
+			const merkleTree = new tree.MerkleTree({ db: this._chainConnectorPluginDB });
+			for (const ccm of ccmHashesList) {
+					merkleTree.append(ccm);
 				}
+			const messageWitness = await merkleTree.generateRightWitness(ccmHashesList.length);
 
-				aggregateBFTWeight += blsKeyToBFTWeight[blsKey];
-			}
+			listOfInboxUpdates.push({
+				crossChainMessages: ccmHashesList,
+				messageWitnessHashes: messageWitness,
+				outboxRootWitness: inclusionProof,
+			});
 		}
 
-		return aggregateBFTWeight >= lastCertificateThreshold;
+		return listOfInboxUpdates;
 	}
 
-	private async _calculateInboxUpdate(sendingChainID: Buffer): Promise<InboxUpdate> {
-		// TODO: Fetch as many CCMs as configured in _ccuFrequency.ccm, between lastCertificateHeight and certificateHeight - After #7569
-		const crossChainMessages = await this._sidechainChainConnectorDB.getCrossChainMessages();
-		const serialzedCCMs = crossChainMessages.map(ccm => codec.encode(ccmSchema, ccm));
-
-		// TODO: should use the store prefix with sendingChainID after issue https://github.com/LiskHQ/lisk-sdk/issues/7631
-		const outboxKey = sendingChainID;
-
-		const stateProveResponse = await this._sidechainAPIClient.invoke<ProveResponse>('state_prove', {
-			queries: [outboxKey],
-		});
-		const inclusionProofOutboxRoot: OutboxRootWitness = {
-			bitmap: stateProveResponse.proof.queries[0].bitmap,
-			siblingHashes: stateProveResponse.proof.siblingHashes,
-		};
-
-		return {
-			crossChainMessages: serialzedCCMs,
-			messageWitnessHashes: [],
-			outboxRootWitness: inclusionProofOutboxRoot,
-		};
+	//
+	private _getListOfCCMs(
+		_crossChainMessages: CrossChainMessagesFromEvents[],
+		_newBlockHeader: BlockHeader,
+	): { ccm: CCMsg, height: number, inclusionProof: OutboxRootWitness }[][] {
+		return [[]];
```

Comment:
Not sure what is happening here. It just returns `[[]]`?

Reply:
its updated now